src/components/ContentPlaceholder/ContentPlaceholder.js

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import bem from 'bem';
+import styles from './ContentPlaceholder.scss';
+
+const { block, elem } = bem({
+    name: 'ContentPlaceholder',
+    classnames: styles,
+    propsToMods: []
+});
+
+const calcDuration = seconds => {
+    if (!seconds) {
+        return null;
+    }
+
+    return `${seconds}s`;
+};
+
+const calcMaskWidth = width => {
+    if (!width) {
+        return null;
+    }
+
+    return {
+        width: `${100 - Math.max(Math.min(width, 100), 0)}%`
+    };
+};
+
+const ContentPlaceholder = props => {
+    const { duration, height, width, ...rest } = props;
+
+    return (
+        <div
+            {...rest}
+            {...block(props)}
+            style={{
+                animationDuration: calcDuration(duration),
+                height
+            }}
+        >
+            &nbsp;
+            <div {...elem('mask', props)} style={calcMaskWidth(width)} />
+        </div>
+    );
+};
+
+ContentPlaceholder.displayName = 'ContentPlaceholder';
+
+ContentPlaceholder.propTypes = {
+    /** Animation duration in seconds */
+    duration: PropTypes.number,
+    /** Custom height in pixels for the content placeholder */
+    height: PropTypes.number,
+    /** Width of the content placeholder, relative to its parent */
+    width: PropTypes.number
+};
+
+ContentPlaceholder.defaultProps = {
+    duration: null,
+    height: null,
+    width: null
+};
```

Comment:
I think it would be easier to read if the defaults would be the actual defaults defined in the css. `duration: 1, width: 0`. Then the `if` can be removed from the calc functions, which makes it simpler (and a bit faster)

Reply:
The problem with that, I think, is that the styles will always be declared inline (even by default). I don't think that's a very good practice in terms of specificity (inline styling will always take precedence).

Reply:
(continuing here)

I don't think we should declare the default both in stylesheet and inline, so then it would be all-inline. I think it depends on what you think is more important:

1. Readability: Keep default and custom value both inline for sake of readability because they're all in one place
2. How your code works: keep default CSS in stylesheet, declare inline only in case of custom values (because CSS should be kept in stylesheets as much as possible, otherwise you may run into unexpected specificity issues etc. And presentation & markup should be separated, ideally)

I personally prefer 2